question22/main.js

```diff
@@ -0,0 +1,188 @@
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTableElement(userData);
+    } else {
+    wrapper.textContent = "Failed to display data.";
+    console.error('Failed to display data.'); 
+  }
+}
+init();
+
+const userDataColumn = {
+  userId: {
+    value : "ID",
+    hasSort : true
+  },
+  name: {
+    value : "名前",
+    hasSort : false
+  },
+  gender: {
+    value : "性別",
+    hasSort : false
+  },
+  age: {
+    value : "年齢",
+    hasSort : true
+  }
+}
+
+const renderTableElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  table.id = "js-table";
+  tbody.id = "js-tbody";
+  wrapper.appendChild(table).appendChild(getTableHeadElement());
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  addClickEventToSortButton();
+}
+
+const getTableHeadElement = () => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  const userDataColumnKeys = Object.keys(userDataColumn);
+
+  userDataColumnKeys.forEach((key) => {
+    const th = createElementWithClassName("th", "th");
+    const value = userDataColumn[key].value;
+    th.textContent = value; 
+    userDataColumn[key].hasSort && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th); 
+  });
+
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const addClickEventToSortButton = () => {
+  const sortButtons = [...document.querySelectorAll(".sortButton")];
```

Comment:
nits
I thought it would be better to add js-prefix.
Although the DOM is created in JS this time, CSS is applied to the same class,
and I thought that changing the class name in JS would lead to bugs.

Reply:
Thank you for your review! I was wondering if it would be better to separate the names since we are operating in JS as you said. Thank you for your valuable input!
fixed commit: https://github.com/csakyo/homework/pull/37/commits/3ab3ca06f88bea3aac611f3fe422639efad0e22d